IV bags and drips redux

comma opened the pull request:
Bloodpack redux  …
Makes them generic IV bags.
Volume lowered to 120u from 200u.
Medbay and blood crate now start with NANOBLOOD - using it would keep more or less same speed/volume of blood replenishment without needing crazy ass bags and hacks to increase transfer amounts by 20x

IV bags can be used to give people IV - drag them onto person and it'll start pumping. Change amount to transfer with a verb like on other reagent containers. It has to be kept in hand to work (anyone's hand).

IV bags can be tampered with by using syringes now.


IV drip refactor  …
Makes it a structure that processes when someone's hooked. Doesn't need to carry all the machinery baggage, it literally was one just for Process().

Can now set the transfer amount from it, between REM, 1 and 2 units per tick. Current value shows on examine.

comma committed:
Bloodpack redux

Makes them generic IV bags.
Volume lowered to 120u from 200u.
Medbay and blood crate now start with NANOBLOOD - using it would keep more or less same speed/volume of blood replenishment without needing crazy ass bags and hacks to increase transfer amounts by 20x

IV bags can be used to give people IV - drag them onto person and it'll start pumping. Change amount to transfer with a verb like on other reagent containers. It has to be kept in hand to work (anyone's hand).

IV bags can be tampered with by using syringes now.

comma committed:
IV drip refactor

Makes it a structure that processes when someone's hooked. Doesn't need to carry all the machinery baggage, it literally was one just for Process().

Can now set the transfer amount from it, between REM, 1 and 2 units per tick. Current value shows on examine.

PsiOmegaDelta commented on code/game/objects/structures/iv_drip.dm:
```diff
@@ -0,0 +1,167 @@
+	attached = null
+	qdel(beaker)
+	beaker = null
+	..()
```
Not returning the result from ..()

PsiOmegaDelta commented on code/game/objects/structures/iv_drip.dm:
```diff
@@ -0,0 +1,167 @@
+		attached = null
+	else if(ishuman(over_object))
+		visible_message("\The [usr] hooks \the [over_object] up to \the [src].")
+		attached = over_object
```
Need to ensure a deleted mob is detached.

comma committed:
Repaths things

comma committed:
Changelog

PurpleMartinJCK commented:
Can you leave a couple of bags as backup anyway?

PurpleMartinJCK commented on code/modules/reagents/reagent_containers/blood_pack.dm:
```diff
@@ -2,59 +2,113 @@
+	volume = 120
+	possible_transfer_amounts = "0.2;1;2"
+	amount_per_transfer_from_this = REM
+	flags = OPENCONTAINER
```
Why did you decide to make this an open container?

comma replied:
So it can be actually filled with other chemicals? It's kinda most of point of the change.
It won't splash or anything if you're worried about that, that's special beaker interactin.

comma commented:
Couple of bags of what? Blood? Nanoblood works exactly same but better, using normal blood would be just frustrating. It'd work in a pinch, but if you want same reblooding speed as now, you gotta use nano.

Dukica99 commented:
So do you mean by this that the rate of blood transfer was changed? Because fucking up someone's immune system does not sound fun if we have to transfer a lot of blood (heck, some cases even had like 20% blood left!)

Having regular O- blood would still be nice, unless you changed the rates.

comma commented:
Rates were indeed changed. And it's not fucking it up that much, no worse than spaceacilline in non OD doses.

comma committed:
Adjusts nanoblood effects

Only do immunity thing if we actually replaced blood.
Lower amounts it wrecks it by, not sure what I was drinking when I set it that high.

Spamcat commented:
rename nanoblood to "RBC mass" for maximum realism

PurpleMartinJCK commented:
The blood would be nice to have in recipies. Using your own blood as a chemist isn't really a good thing.

comma commented:
Use someone else's?
If chemist is too scared to donate blood and no one is willing, they can order more stuff from cargo.

Techhead0 commented:
Or drain some monkeys.

PurpleMartinJCK commented:
It's not about being scared to donate blood, it's unprofessional to be pulling and using your own blood straight from the vein for chemistry.

comma commented:
1.Blood donations exist.
2.This PR didn't add nanoblood, why are you bringing this up here.
